### PR TITLE
web/satellite: Buckets page rework

### PR DESCRIPTION
**Pull request by NikolaiYurchenko**
What: 
Refactored Buckets page in satellite web interface
Why:
Reworked with implementation of newly created common components such Header and Pagination

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: web/satellite/src/components/buckets/BucketArea.vue [hunk 1]
```diff
@@ -7,18 +7,16 @@
         <div class="buckets-overflow" v-else>
             <div class="buckets-header">
                 <p>Buckets</p>
-                <SearchArea/>
+                <HeaderComponent class="buckets-header-component" placeholder="Buckets" :search="fetch"/>
```
Comment (VinozzZ) [inline web/satellite/src/components/buckets/BucketArea.vue]: Should this placeholder be passed in as a prop?

Comment (crawter) [reply web/satellite/src/components/buckets/BucketArea.vue]: yeah, we want to have different placeholders in our header component. So we use it as a prop

File: web/satellite/src/components/buckets/BucketArea.vue [hunk 2]
```diff
@@ -28,22 +26,24 @@
...
         components: {
             EmptyState,
-            SearchArea,
             SortingHeader,
             BucketItem,
-            PaginationArea,
             NoBucketArea,
+            HeaderComponent,
+            Pagination,
+            List
```
Comment (crawter) [inline web/satellite/src/components/buckets/BucketArea.vue]: trailing coma

File: web/satellite/src/components/buckets/BucketArea.vue [hunk 3]
```diff
@@ -53,60 +53,68 @@
...
 
+        public get totalPageCount(): number {
+            return this.$store.state.bucketUsageModule.page.pageCount;
+        }
+
         public get totalCountOfBuckets(): number {
             return this.$store.state.bucketUsageModule.totalCount;
         }
 
+        public get itemComponent() {
```
Comment (crawter) [inline web/satellite/src/components/buckets/BucketArea.vue]: return type

File: web/satellite/src/components/buckets/BucketItem.vue [hunk 4]
```diff
@@ -17,58 +17,40 @@
     @Component
     export default class BucketItem extends Vue {
         @Prop()
```
Comment (crawter) [inline web/satellite/src/components/buckets/BucketItem.vue]: let's add default () => new BucketUsage();

**Pull request merged**